presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -141,6 +144,11 @@ bool cachePeriodicPersistenceEnabled() {
       std::chrono::seconds::zero();
 }
 
+bool isSharedLibrary(const fs::path& path) {
+  std::string pathExt = path.extension().string();
```

Comment:
nit: const.
We should also convert pathExt to lower-case to handle upper-case extensions.